backend/src/v5/middleware/dataConverter/schemas/groups.js

```diff
@@ -0,0 +1,148 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { isNumber, isString } = require('../../../utils/helper/typeCheck');
+const Yup = require('yup');
+const { types } = require('../../../utils/helper/yup');
+
+const Group = {};
+
+const objectEntryValidator = Yup.object().shape({
+	account: types.strings.username.required(),
+	model: types.id.required(),
+	shared_ids: Yup.array().of(types.id).min(1).optional(),
+	ifc_guids: Yup.array().of(Yup.string().length(22)).min(1).optional(),
+}).test(
+	'Object item check',
+	'Can only contain either ifc_guids or shared_ids',
+	(value) => (value.shared_ids && !value.ifc_guids) || (!value.shared_ids && value.ifc_guids),
+);
+
+const operators = {
+	IS_EMPTY: 0,
+	IS_NOT_EMPTY: 0,
+	IS: 1,
+	IS_NOT: 1,
+	CONTAINS: 1,
+	NOT_CONTAINS: 1,
+	REGEX: 1,
+	EQUALS: 1,
+	NOT_EQUALS: 1,
+	GT: 1,
+	GTE: 1,
+	LT: 1,
+	LTE: 1,
+	IN_RANGE: 2,
+	NOT_IN_RANGE: 2,
+};
+
+const ruleParametersTypeCheck = (operator, values) => {
+	switch (operator) {
+	case 'IS':
+	case 'IS_NOT':
+	case 'CONTAINS':
+	case 'NOT_CONTAINS':
+	case 'REGEX':
+		return values.every(isString);
+	case 'EQUALS':
+	case 'NOT_EQUALS':
+	case 'GT':
+	case 'GTE':
+	case 'LT':
+	case 'LTE':
+		return values.every(isNumber);
+	default:
+		// range checks
+		return values.length % 2 === 0 && values.every(isNumber);
+	}
```

Comment:
This works and looks good, but we could also do the same with `if..else`. I guess it only becomes a problem if someone edits the order in a wrong way.

Reply:
I think it's pretty common understanding that case statement falls through unless you returned/ break?

Reply:
I can change it to an if, if that's more obvious